### PR TITLE
Documentation: Clarify deprecations documentation.

### PR DESCRIPTION
**Pull request by glendaviesnz**
## Description

The current deprecations documentations is not 100% clear and easy to follow.

## Testing Instructions
Read and see if it makes sense 😄

### CODE REVIEW

File: docs/reference-guides/block-api/block-deprecation.md [hunk 1]
```diff
@@ -7,9 +7,17 @@ When updating static blocks markup and attributes, block authors need to conside
...
+4. The attributes, and any innerBlocks, from the first deprecation to generate a valid block are then passed back to the current `save` method to generate new valid content for the block.
+5. At this point the current block should now be in a valid state and the deprecations workflow stops.
+
+It is important to note that if a deprecation's `save` method does not produce a valid block then it is skipped completely, including its `migrate` method, even if `isEligible` would return true for the given attributes. This means that if you have several deprecations for a block and want to perform a new migration, like moving content to `InnerBlocks`, you may need to update the `migrate` methods in multiple deprecations in order for the required changes to be applied to all previous versions of the block.
+
+It is also important to note that if a deprecations `save` method imports additional methods you may want to add a snapshot copy of these methods to the deprecations file instead of importing them in order to avoid future changes to these methods inadvertently breaking the deprecations.
```
Comment (andrewserong) [inline docs/reference-guides/block-api/block-deprecation.md]: I found this sentence a little hard to parse. Is it worth breaking it up into a couple of sentences?

```suggestion
It is also important to note that if a deprecation's `save` method imports additional functions from other files, changes to those files may accidentally change the behavior of the deprecation. You may want to add a snapshot copy of these functions to the deprecations file instead of importing them in order to avoid inadvertently breaking the deprecations.
```

**Review by andrewserong — COMMENTED** [review]
Thanks for tidying up and clarifying the documentation @glendaviesnz, this looks like a good improvement! It makes it much clearer splitting out into an ordered list 👍

I've just left a comment about one of the sentences, but other than that, this LGTM!

File: docs/reference-guides/block-api/block-deprecation.md (hunk 1 above)
Comment (glendaviesnz) [reply docs/reference-guides/block-api/block-deprecation.md]: yeh, that is clearer - thanks.

**Review by ramonjd — APPROVED** [review]
Thanks for the effort to make this document easier to read @glendaviesnz 

Down the line, I think the docs could be made even clearer with illustrations or demonstration code documenting the update flow.

Also, I think the definitions might be better placed somewhere at the top of the document, before they're used in full sentences. 

Deprecations aren't immediately accessible to newcomers at the best of times, so anything we can do to improve this document is okay by me.

🍺

**Review by andrewserong — APPROVED** [review]
Thanks for updating! LGTM 👍

**Review by apeatling — APPROVED** [review]
This is much clearer. 👍
